Dockerfile

```diff
@@ -12,16 +12,18 @@ COPY go.sum go.sum
 RUN go mod download
 
 # Copy the go source
-COPY cmd/main.go cmd/main.go
+COPY cmd/ cmd/
 COPY api/ api/
 COPY internal/controller/ internal/controller/
+COPY ext/ ext/
+COPY pkg/ pkg/
 
 # Build
 # the GOARCH has not a default value to allow the binary be built according to the host where the command
 # was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
 # the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
 # by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
-RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager ./cmd
 
```

Comment:
This Dockerfile is quite different from the one in master. I think we'll need to keep it close to the master as much as we can in order to make it easier to merge into master in the future.

Reply:
To avoid misunderstanding, let's keep the original Dockerfile as is and I created a new controller.Dockerfile that can be used later to run a controller.

Reply:
now that we're not using the original Dockerfile for the controller, shall we restore `Dockerfile` to its original content prior to the start of crd work?

Reply:
hm, the Dockerfile in this PR is exactly the same as Dockerfile in master https://github.com/argoproj-labs/argocd-image-updater/blob/master/Dockerfile. It differs from the current version in crd branch but the same as in master.